src/Symfony/Component/HttpClient/Response/ResponseTrait.php

```diff
@@ -253,7 +253,7 @@ private static function initialize(self $response): void
     private static function addResponseHeaders(array $responseHeaders, array &$info, array &$headers, string &$debug = ''): void
     {
         foreach ($responseHeaders as $h) {
-            if (11 <= \strlen($h) && '/' === $h[4] && preg_match('#^HTTP/\d+(?:\.\d+)? ([12345]\d\d) .*#', $h, $m)) {
+            if (11 <= \strlen($h) && '/' === $h[4] && preg_match('#^HTTP/\d+(?:\.\d+)?\s([12345]\d\d).*#', $h, $m)) {
```

Comment:
should be `'#^HTTP/\d+(?:\.\d+)? ([12345]\d\d)(?: |$)#'` to me

Reply:
Why you wish do more complex?. 

By me are ok while bug is fixed.

Reply:
It's not more complex, it's stricter. Thanks :)

Reply:
Well, also are valid and same stricter:

`#^HTTP/\d+(?:\.\d+)?\s([12345]\d\d)\s?$#`

One question more, why you write "whitespace" literal instead of use \s meta-sequencer to mark when you expect a whitespace on regexp?

Thanks

Reply:
Because we don't want to accept any whitespace.
E.g. a \r or a \n would be a security issue, and a \t in not compliant.
The HTTP RFC precisely defines that only spaces are allowed here, we must not accept more.

Reply:
Ah ok, thanks for the explain. Yes, are correct and my fail when i put `\s?$` I thought it was the same.

:)

Reply:
Need you a new PR with the correct expression?

Reply:
You can update this one!